release_dashboard/lib/widgets/create_release_substeps.dart

```diff
@@ -33,9 +34,35 @@ class CreateReleaseSubsteps extends StatefulWidget {
 }
 
 class CreateReleaseSubstepsState extends State<CreateReleaseSubsteps> {
-  // Initialize a public state so it could be accessed in the test file.
+  // Initialize a public state so it could be accessed in the test file
   @visibleForTesting
   late Map<String, String?> releaseData = <String, String?>{};
+  // When isEachInputValid['substep1'] is true, 'substep1' is valid. If it false, it is invalid
+  @visibleForTesting
+  Map<String, bool> isEachInputValid = <String, bool>{};
+
+  @override
+  void initState() {
+    // engine cherrypicks, framework cherrypicks and dart revision are optional and valid with empty input at the beginning
+    for (final String substep in CreateReleaseSubsteps.substepTitles) {
+      if (<String>[
```

Comment:
This would be cleaner by initializing the list of constants outside of the conditional